src/test_report.py

```diff
@@ -18,9 +22,97 @@ class TestReport:
+
+        if self._email_config['in_reply_to']:
+            email_msg["In-Reply-To"] = email_msg["References"] = self._email_config['in_reply_to']
+
+        if is_ascii(self._email_config['subject']):
+            email_msg['Subject'] = self._email_config['subject']
+        else:
+            email_msg['Subject'] = Header(self._email_config['subject', "utf-8"]).encode()
+
+        email_msg['To'] = u", ".join(self._email_config['send_to'])
+
+        email_msg['From'] = os.getenv('MAIL_FROM')
+
+        email_msg["Cc"] = u", ".join(self._email_config['cc_addrs'])
+
+        gen.flatten(email_msg)
+
+        return email_msg
+
+
+    def send_email(self, template):
+        """Send emails to the specific email"""
+
+        errors = []
+        status = models.ERROR_STATUS
+
+        email_msg = self.create_email(template)
+
+        MAIL_USERNAME = os.getenv('MAIL_USERNAME'),
+        MAIL_PASSWORD = os.getenv('MAIL_PASSWORD'),
+        MAIL_FROM = os.getenv('MAIL_FROM'),
+        MAIL_PORT = int(os.getenv('MAIL_PORT')),
+        MAIL_HOST = os.getenv('MAIL_HOST'),
+
+        if all([MAIL_FROM, MAIL_HOST]):
+            server = None
+
+            try:
+                if MAIL_PORT == 465:
+                    server = smtplib.SMTP_SSL(MAIL_HOST, port=MAIL_PORT)
+                else:
+                    server = smtplib.SMTP(MAIL_HOST, port=MAIL_PORT)
+
+                if all([MAIL_USERNAME, MAIL_PASSWORD]):
+                    server.login(MAIL_USERNAME, MAIL_PASSWORD)
+
+                server.sendmail(MAIL_FROM, self._email_config['send_to'], email_msg)
+                status = models.SENT_STATUS
+            except (smtplib.SMTPAuthenticationError, smtplib.SMTPConnectError), ex:
+                self._print("SMTP conn/auth error")
+                errors.append((ex.smtp_code, ex.smtp_error))
+            except (smtplib.SMTPRecipientsRefused, smtplib.SMTPSenderRefused), ex:
+                self._print("Error sending email: recipients or sender refused")
+                errors.append((ex.smtp_code, ex.smtp_error))
+            except (smtplib.SMTPHeloError, smtplib.SMTPDataError), ex:
+                self._print("SMTP server error")
+                errors.append((ex.smtp_code, ex.smtp_error))
+            except smtplib.SMTPException, ex:
+                self._print("Generic SMTP error: no auth method, ...")
+                errors.append((ex.smtp_code, ex.smtp_error))
+            except Exception, ex:
+                self._print(ex)
+                self._print("Unexpected SMTP error: %s", str(ex))
+                errors.append((500, str(ex)))
+            finally:
+                if server is not None:
+                    server.quit()
+        else:
+            errors.append(
+                (500, "No STMP host and/or sender specified: no email sent"))
+            self._print("Cannot send emails: no SMTP host and/or sender specified")
+
+        return status, errors
```

Comment:
Why do we need `status` and `error` values. They don't seem to be used anywhere.